### ISSUE TITLE
Test and update skorch notebooks on Colab

### ISSUE THREAD

**Issue by BenjaminBossan** [skorch-dev/skorch#919]
I was notified that the command we use in our notebooks to install packages on Colab no longer works:

`! [ ! -z "$COLAB_GPU" ] && pip install ...`

The problem is that apparently, the `$COLAB_GPU` env var was removed. I did a quick check and maybe another env var could be used to check, but who knows if that would be robust. Another approach would be something along the lines of:

```python
try:
    import google.colab
    subprocess.run(['python', '-m', 'pip', 'install', ...])
except ImportError:
    pass
```

but I haven't tested that.

Other than these issues, it would be good to check if these notebooks still run successfully.

**Comment by sawradip** [comment]
I would like to do this.

**Comment by BenjaminBossan** [comment]
Thanks a lot. As mentioned, there are two parts to this task:

1. Find an alternative to the previous install "hack" which doesn't work anymore.
2. Make sure that the rest of the notebook still runs.

Would you like to work on one or both? Regarding point 1, do you have a suggestion how to solve it? As I rarely use colab myself, I don't have a good idea.

**Comment by sawradip** [comment]
For point 1,  I find that running simple `! pip install skorch` installs the package successfully.  Can you let me know if I am missing something? I have been using Colab for about a year now, and I have been installing packages in that way. Did you use that hack for any specific reason? or it was just a requirement for that time to install new packages on colab.

```
! pip install skorch
```
and in another cell,

```
from skorch import NeuralNetClassifier
```

And though I am new to skorch, I would like to try the 2nd task also.

**Comment by BenjaminBossan** [comment]
> I find that running simple `! pip install skorch` installs the package successfully. Can you let me know if I am missing something?

Yeah, the reason why we don't want to do this is because the same notebook is used by people who don't use colab. E.g. when I run the notebook locally, I don't want to run any pip installs. Therefore. we want to run the install only when in a colab environment.

> And though I am new to skorch, I would like to try the 2nd task also.

Please go ahead. If you find any issues, just ask.

**Comment by BenjaminBossan** [comment]
Great, thanks.

We can assume that users running locally don't want to install stuff when running the notebook, so that step should just be skipped.

**Comment by BenjaminBossan** [comment]
Ah yes, I see. In the definition of the `MyTensorBoard` class, the `on_batch_begin` method needs to be changed a little bit:

```python
        def on_batch_begin(self, net, batch, **kwargs):
            X, y = batch
            if self.first_batch_:
                # only add graph on very first batch
                self.add_graph(net.module_, X)
```

I made this change it and worked.

Furthermore, I see that in that notebook, the installation says

> pip install torch scikit-learn==0.21.* skorch

But I think this should be changed to

> pip install torch torchvision skorch

**Comment by sawradip** [comment]
It doesn't need scikit-learn anymore?

**Comment by sawradip** [comment]
I am a little sick..and I will update soon.

**Comment by BenjaminBossan** [comment]
@sawradip Any updates? Feel free to ask if you need more information or help.

**Comment by sawradip** [comment]
Hello, Sorry for being late. I have resumed my work. 

And I am going to add a commit today. I will let you know if I face any more issue.

**Comment by sawradip** [comment]
I have added a pull request. 

Could you check it and let me know?

**Comment by sawradip** [comment]
Thank you for your reviews, and I am fixing those. I thing I have faced and think should be fixed is :

![Screenshot (3)](https://user-images.githubusercontent.com/67541368/210774075-83677f1b-f083-4f82-ad00-04c429b78175.png)

If GPU is connected. This error arises. 

I am seeing, changing this line
``` samples = samples.detach().numpy()   ```
to this line
``` samples = samples.cpu().detach().numpy()   ```
will make it work for both CPU and GPU.

And as you asked if I changed anything in those notebooks, I don't remember changing anything that effects the code. One or two comments might have been added, but as some were already there, it is tough for me also to figure out as `diff` is not a particularly good view for notebook. You can be sure taking a look at the comments.

**Comment by BenjaminBossan** [comment]
@sawradip Could you please post your reply on the PR (#928) instead of here?

**Comment by sawradip** [comment]
Yeah...did.

**Comment by BenjaminBossan** [comment]
Closed via #928
